### PR TITLE
[Photos] Update for Xcode10-beta2.

### PR DESCRIPTION
**Pull request by mandel-macaque**
This change also includes PhotosUI needed updates.

### CODE REVIEW

File: src/Photos/Enums.cs [hunk 1]
```diff
@@ -337,4 +337,18 @@ public enum PHLivePhotoEditingError : long {
 		Unknown,
 		Aborted,
 	}
+
+	[Mac (10,14, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	public enum FigEXIFCustomRenderedValue : short {
```
Comment (spouliot) [inline src/Photos/Enums.cs]: 4 (3+) letter acronym -> `Exif`

File: src/Photos/Enums.cs [hunk 2]
```diff
@@ -337,4 +337,18 @@ public enum PHLivePhotoEditingError : long {
...
+		NotCustom = 0,
+		Custom = 1,
+		HDRImage = 2,
+		HDRPlusEV0_HDRImage = 3,
+		HDRPlusEV0_EV0Image = 4,
+		PanoramaImage = 6,
+		SDOFImage = 7,
+		SDOFPlusOriginal_SDOFImage = 8,
+		SDOFPlusOriginal_OriginalImage = 9,
```
Comment (spouliot) [inline src/Photos/Enums.cs]: same above ^^^ `HDR`, `SDOF`

File: src/photosui.cs [hunk 3]
```diff
@@ -198,6 +210,14 @@ interface PHProjectExtensionController {
 		[Abstract]
 		[Export ("finishProjectWithCompletionHandler:")]
 		void FinishProject (Action completion);
+
+		[Protected]
+		[Mac (10,14, onlyOn64: true)]
+		[Export ("typeDescriptionDataSourceForCategory:invalidator:")]
+		IPHProjectTypeDescriptionDataSource GetTypeDescriptionDataSource (string category, IPHProjectTypeDescriptionInvalidator invalidator);
```
Comment (dalexsoto) [inline src/photosui.cs]: `string category` to `NSString`

File: src/photosui.cs [hunk 4]
```diff
@@ -227,6 +247,29 @@ interface PHProjectTypeDescription : NSSecureCoding {
 
 		[Export ("initWithProjectType:title:description:image:")]
 		IntPtr Constructor (NSString projectType, string localizedTitle, [NullAllowed] string localizedDescription, [NullAllowed] UIImage image);
+
+		[Mac (10,14, onlyOn64: true)]
+		[Export ("initWithProjectType:title:attributedDescription:image:subtypeDescriptions:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSString projectType, string localizedTitle, [NullAllowed] NSAttributedString localizedAttributedDescription, [NullAllowed] UIImage image, PHProjectTypeDescription[] subtypeDescriptions);
```
Comment (dalexsoto) [inline src/photosui.cs]: Do we know what `NSString projectType` is? Is it a smart enum? can we make a better API here? if so you might be able to use BindAs

File: src/photosui.cs [hunk 5]
```diff
@@ -326,5 +397,82 @@ interface PHProjectSectionContent : NSSecureCoding {
...
+	interface IPHProjectTypeDescriptionDataSource {}
+	[Mac (10,14, onlyOn64: true)]
+	[NoiOS][NoTV]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface PHProjectTypeDescriptionDataSource
+	{
+		[Abstract]
+		[Export ("subtypesForProjectType:")]
+		PHProjectTypeDescription[] GetSubtypes (string projectType);
```
Comment (dalexsoto) [inline src/photosui.cs]: I think we might want NSString here instead, even more if this is a smart enum

File: src/photosui.cs [hunk 6]
```diff
@@ -326,5 +397,82 @@ interface PHProjectSectionContent : NSSecureCoding {
...
+
+	interface IPHProjectTypeDescriptionInvalidator {}
+	[Mac (10,14, onlyOn64: true)]
+	[NoiOS][NoTV]
+	[Protocol]
+	interface PHProjectTypeDescriptionInvalidator
+	{
+		[Abstract]
+		[Export ("invalidateTypeDescriptionForProjectType:")]
+		void InvalidateTypeDescription (string projectType);
```
Comment (dalexsoto) [inline src/photosui.cs]: same, we might want NSString here if it is a smart enum value

File: src/photosui.cs [hunk 7]
```diff
@@ -326,5 +397,82 @@ interface PHProjectSectionContent : NSSecureCoding {
...
+
+		[Abstract]
+		[Export ("invalidateFooterTextForSubtypesOfProjectType:")]
+		void InvalidateFooterTextForSubtypes (string projectType);
+	}
+
+	[iOS (8,0)]
+	[NoMac][NoTV]
+	[BaseType (typeof (NSExtensionContext))]
+	interface PHEditingExtensionContext
```
Comment (dalexsoto) [inline src/photosui.cs]: `[DisableDefaultCtor]` I do not think this is meant to be created

**Comment by mandel-macaque** [comment]
As talked with @dalexsoto  and @spouliot we will move the projectType params to NSString since there are no more values than "Undefined" and we do not know how they are compared.

File: src/Photos/Enums.cs [hunk 8]
```diff
@@ -340,15 +340,15 @@ public enum PHLivePhotoEditingError : long {
...
-		HDRPlusEV0_EV0Image = 4,
+		HdrImage = 2,
+		HdrPlusEV0_HDRImage = 3,
+		HdrPlusEV0_EV0Image = 4,
 		PanoramaImage = 6,
-		SDOFImage = 7,
-		SDOFPlusOriginal_SDOFImage = 8,
-		SDOFPlusOriginal_OriginalImage = 9,
+		SdofImage = 7,
+		SdofPlusOriginal_SDOFImage = 8,
```
Comment (dalexsoto) [inline src/Photos/Enums.cs]: `SDOFImage` -> `SdofImage`

Comment (mandel-macaque) [reply src/Photos/Enums.cs]: Agh, the second one, well spotted.

File: src/Photos/Enums.cs [hunk 9]
```diff
@@ -337,4 +337,18 @@ public enum PHLivePhotoEditingError : long {
...
 		Aborted,
 	}
+
+	[Mac (10,14, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	public enum FigExifCustomRenderedValue : short {
+		NotCustom = 0,
+		Custom = 1,
+		HdrImage = 2,
+		HdrPlusEV0_HDRImage = 3,
```
Comment (spouliot) [inline src/Photos/Enums.cs]: `*_HdrImage`

File: src/photos.cs [hunk 10]
```diff
@@ -1267,8 +1271,13 @@ interface PHProjectChangeRequest {
 		[Export ("projectExtensionData", ArgumentSemantic.Copy)]
 		NSData ProjectExtensionData { get; set; }
 
+		[Deprecated (PlatformName.MacOSX, 10, 14)]
```
Comment (chamons) [inline src/photos.cs]: Any header comment on what to use instead?

Comment (mandel-macaque) [reply src/photos.cs]: None :(

**Comment by mandel-macaque** [comment]
@spouliot let me know if you are ok with merging this.

**Pull request merged**